### PR TITLE
cmake: split cdio and remove iso9660

### PR DESCRIPTION
**Pull request by lrusak**
~~This splits cdio and iso9660 into two separate cmake modules. They are shipped by the same package (cdio) but we only require cdio (and not iso9660) on windows. Both are required on all other platforms.~~

This makes it more clear that two separate libraries are needed.

This change will allow (with some future work) to make these packages optional.

This also bumps the required version to 0.80 which is the lower version we ship in our depends. No idea if it works with an earlier version (but we shouldn't care anyways).

EDIT: This now drops support for iso9660 as it isn't needed at all.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -131,6 +130,10 @@ set(required_deps ASS
                   ZLIB
                   ${PLATFORM_REQUIRED_DEPS})
 
+if(NOT CORE_SYSTEM_NAME STREQUAL windows AND NOT CORE_SYSTEM_NAME STREQUAL windowsstore)
+  list(APPEND required_deps Iso9660)
+endif()
```
Comment (Rechi) [inline CMakeLists.txt]: no platform hack in common CMake files

**Review by Rechi — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (lrusak) [reply CMakeLists.txt]: says who? There is other platform specifics in there already

**Comment by Paxxi** [comment]
So to avoid platform hacks we should build iso9660 as a separate lib and not only baked into the cdio dll?

I have no strong opinions on this, if windows still builds and works then it's fine by me 😀

**Comment by Rechi** [comment]
If the upstream build system always generates two libs and the functionality of both is needed, the same has to be done one windows.

**Comment by Paxxi** [comment]
What upstream does in their build systems is irrelevant as we mostly don't use them. Usually it makes sense to keep fairly close to upstream but there's no rule about.

**Review by Rechi — CHANGES_REQUESTED** [review]
The iso9660 library is currently need, or is the `cdio/iso9660.h` include at the following place not required?

https://github.com/xbmc/xbmc/blob/48e6c7666358f31c0b78a33494e688cf04d0c49a/xbmc/filesystem/iso9660.h#L22

**Comment by lrusak** [comment]
@Rechi better now?

**Comment by Rechi** [comment]
If types from cdio (iso9660) are used, include the upstream header to get the correct definition. If you don't want to include it, use a common type.

**Comment by lrusak** [comment]
> If types from cdio (iso9660) are used, include the upstream header to get the correct definition. If you don't want to include it, use a common type.

That doesn't really make sense. Plus the code is dead with #16262 anyways

**Comment by Rechi** [comment]
If #16262 removes the code, better get that merged first and then have a look at this PR again, if there is still something left.

**Pull request merged**